subprojects/core/src/integTest/groovy/org/gradle/api/tasks/AbstractIncrementalTasksIntegrationTest.groovy

```diff
@@ -401,4 +421,11 @@ abstract class AbstractIncrementalTasksIntegrationTest extends AbstractIntegrati
         """
         succeeds("incrementalCheck")
     }
+
+    void maybeExpectIncrementalTaskInputsDeprecationWarning(String className = 'BaseIncrementalTask', String methodName = 'execute') {
+        String source = "${className}.${methodName}"
+        if (expectedIncrementalTaskInputsDeprecationWarning && incrementalTaskInputsDeprecationWarningEmittedFor.add(source)) {
+            executer.expectDocumentedDeprecationWarning """IncrementalTaskInputs has been deprecated. This is scheduled to be removed in Gradle 8.0. On method '$source' use 'org.gradle.work.InputChanges' instead. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_7.html#incremental_task_inputs_deprecation"""
+        }
+    }
```

Comment:
Why do we have this if here? Can't we have different implementations for this method instead in `IncrementalInputsIntegrationTest` vs `IncrementalTaskInputsIntegrationTest` instead?

Reply:
Good idea! I did this and the code became simpler.